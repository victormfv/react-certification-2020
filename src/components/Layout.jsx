import React from 'react';
// import Header from './Header.component';
import './Layout.styles.css';
import Navigation from './Navigation';

function Layout({ children }) {
  return (
    <div>
      <Navigation />
      <main className="container">{children}</main>
    </div>
  );
}

export default Layout;
