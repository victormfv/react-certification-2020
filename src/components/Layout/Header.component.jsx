import React from 'react';
import styled from 'styled-components';
import SwitchButton from './Switch.component';

const Avatar = styled.svg `
        background-color: lightgray;
        width: 40px;
        height: 40px;
        overflow: hidden;
        border-radius: 50%;
        margin-left:20px;
    `;

    const Mag = styled.svg `
        width: 1em;
        height: 1em;
        font-size:1.5rem;
        margin: 2px 5px;
    `;

    const Menu = styled.svg `
        width: 1em;
        height: 1em;
        font-size:1.5rem;
        margin: 0 10px;
    `;

    const Nav = styled.div `
        background-color: #1C5476;
        color:#FFF;
        transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
        box-shadow: 0px 2px 4px -1px rgb(0 0 0 / 20%), 0px 4px 5px 0px rgb(0 0 0 / 14%), 0px 1px 10px 0px rgb(0 0 0 / 12%);
    `;

    const NavHeader = styled.div `
        padding: 4px 20px;
        width: 100%;
        display: flex;
        align-items:center;
        justify-content: space-between;
        margin: 0 auto;
        `;

    const NavLeft = styled.div `
        display: flex;
        align-items:center;
        width:400px;
        @media (min-width: 768px) {
        width:40%;
        }
    `;

    const SearchInput = styled.div `
        width:100%;
        margin-left:20px;
        height:34px;
        border-radius: 3px;
        display:flex;
        align-items:center;
        color: #FFF;
        font-size: 14px;
        font-weight: 300;
        background-color: rgba(255, 255, 255, 0.15);
    `;

    const Input = styled.input `
        font-size: 16px;
        border:none;
        height:1.8em;
        background-color:transparent;
        :active,
        &:focus {
        outline: none;
        box-shadow: none;
        caret-color:white;
        }
        ::placeholder,
        ::-webkit-input-placeholder {
        color: rgba(255, 255, 255, 0.5);

        }
        :-ms-input-placeholder {
        color: rgba(255, 255, 255, 0.5);

        }
    `;

    const NavRight = styled.div `
        align-items:center;
        display:none;
        @media (min-width: 768px) {
        display:flex;
        }
        overflow:hidden;
        right:0;
        svg {
        margin: 0 10px;
        }

    `;

    const MenuLink = styled.a `
    color:white;`;

function Header() {

    

return (
<div>
    <Nav>
        <NavHeader>
            
            <NavLeft>
                <MenuLink href="#">
                    <Menu fill="currentColor" viewBox="0 0 24 24" aria-hidden="true">
                        <path d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"></path>
                    </Menu>
                </MenuLink>
                <SearchInput>
                    <Mag width="24" height="24" fill="none">
                        <path d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z" stroke="currentColor" strokeWidth="2"
                            strokeLinecap="round" strokeLinejoin="round"></path>
                    </Mag>
                    <Input type="text" placeholder="Search" />
                </SearchInput>
            </NavLeft>

            <NavRight>
                <MenuLink href="#">
                    <SwitchButton text="Dark Mode" />
                </MenuLink>
                <MenuLink href="#">
                    <Avatar viewBox="0 0 24 24" fill="currentColor" aria-hidden="true">
                        <path
                            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z">
                        </path>
                    </Avatar>
                </MenuLink>
            </NavRight>
        </NavHeader>
    </Nav>
</div>
)
};

export default Header;