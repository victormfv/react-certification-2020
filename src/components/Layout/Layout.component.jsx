import React from 'react';
import Header from './Header.component';
import './Layout.styles.css';

function Layout({ children }) {

return (
  <div>
    <Header/>
    <main className="container">{children}</main>
  </div>
);

}

export default Layout;
