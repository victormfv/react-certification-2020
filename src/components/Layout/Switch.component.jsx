import React from 'react';
import styled from 'styled-components';

function SwitchButton(props) {
    const Switch = styled.label `
    position: relative;
  display: inline-block;
  width: 60px;
  height: 34px;
  input { 
  opacity: 0;
  width: 0;
  height: 0;
  
}
`;

    const Slider = styled.span `

    position: absolute;
  cursor: pointer;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: #11344A;
  -webkit-transition: .4s;
  transition: .4s;
  border-radius: 34px;
  margin-right:10px;
    &:before {
    position: absolute;
    content: "";
    height: 26px;
    width: 26px;
    left: 4px;
    bottom: 4px;
    background-color: white;
    -webkit-transition: .4s;
    transition: .4s;
    border-radius: 50%;
    }
    

`;
    return ( 
        <div>
            <Switch>
                <input type="checkbox" checked />
                <Slider />
            </Switch> { props.text } 
        </div>
    )
};

export default SwitchButton;