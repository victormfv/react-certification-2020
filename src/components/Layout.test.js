import React from 'react';
import renderer from 'react-test-renderer';
import Layout from './Layout';
import { render, screen } from '@testing-library/react';

describe('Layout Component', () => {

    it('renders correctly', () => {
      const tree = renderer.create(<Layout/>).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('child components render correct', () => {
      render(<Layout/>);
      const i = screen.getByText(/Dark Mode/i);
      expect(i).toBeInTheDocument();
    });
    
  });