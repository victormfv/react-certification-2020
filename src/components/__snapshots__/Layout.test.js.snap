// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component renders correctly 1`] = `
<div>
  <nav
    className="bg-lightblue-850 w-full flex relative justify-between items-center mx-auto px-8 h-20 text-white"
  >
    <div
      className="flex"
    >
      <a
        className="text-white w-16"
        href="/"
      >
        <img
          alt="Menu"
          src="burger.svg"
        />
      </a>
    </div>
    <form
      className="w-full"
      onSubmit={[Function]}
    >
      <div
        className="flex-auto"
      >
        <div
          className="flex pl-2 md:w-96 bg-opacity-20 bg-white rounded px-1  py-1"
        >
          <div
            className="flex items-center justify-between  h-8 w-8"
          >
            <img
              alt="Profile"
              src="magnifying.svg"
            />
          </div>
          <input
            className="placeholder-opacity-50 w-full bg-transparent h-8 text-lg focus:outline-none  focus:border-transparent"
            name="searchInput"
            onChange={[Function]}
            placeholder="Search..."
            type="text"
          />
        </div>
      </div>
    </form>
    <div
      className="md:flex hidden justify-around w-48 "
    >
      <label
        className="flex items-center"
        htmlFor="input"
      >
        <input
          className="relative w-10 h-4 top-0.5 transition-all duration-200 ease-in-out bg-gray-400 rounded-full shadow-inner outline-none appearance-none "
          type="checkbox"
        />
        <span
          className="ml-2"
        >
          Dark Mode
        </span>
      </label>
      <div
        className="bg-gray-350 rounded-full h-11 w-11 flex items-center justify-center"
      >
        <a
          className="text-white"
          href="/"
        >
          <img
            alt="Profile"
            src="avatar.svg"
          />
        </a>
      </div>
    </div>
  </nav>
  <main
    className="flex"
  />
</div>
`;
